### PR TITLE
feat(prisma): use new Adapter interface

### LINKED ISSUE #38: uuid or cuid for Prisma adaptor model IDs
I would like to use `cuid` for model IDs in our Prisma schema.

This will require changing the type of the field.

This in turn forces some of the TS types to change, e.g. `linkAccount` function `userId` paramter changes from `number` to `string`.

Would it be acceptable for force Next Auth Prisma adaptor users to use `String` type for model IDs?

Or is there an easy way to support both `Int` based IDs and `String` based model IDs?

### LINKED ISSUE #70: [prisma] remove unreliable `sessionCache` and `userCache`, as they don’t work in serverless environments
When lambda function concurrency increases, both `sessionCache` and userCache gets out of sync with the database.

Imagine there are 2 concurrent functions, Alice and Bob, running at the same time:

1. Alice creates a session
2. Bob retrieves the session Alice has just created from the database and caches it in its memory
3. The user logs out and thus, Alice revokes the session by removing it from the database and its own cache
4. Bob still has the given user’s session in its cache, and there’s no way to invalidate it without accessing a synchronization node like a database 😕

I would like to suggest removing the Prisma adapter’s cache in its entirely for now, as the resulting issues tend to cause request timeouts on serverless deployment platforms like Vercel.

### PR DESCRIPTION
**Pull request by balazsorban44**
Currently working out the `id: number` thing...

(#46)

Fixes #70, fixes #38

### CODE REVIEW

**Comment by kripod** [comment]
I’ve created a new adapter which works well with string IDs and can be used together with next-auth@3.18.0 out of the box: https://gist.github.com/kripod/35ccf5319c8477de6f71c1877c7de320

File: packages/prisma/prisma/schema.prisma [hunk 1]
```diff
@@ -7,7 +7,7 @@ generator client {
 }
 model Account {
   id                 Int       @id @default(autoincrement())
```
Comment (kripod) [inline packages/prisma/prisma/schema.prisma:9]: I think this should also be converted into a string after all.

File: packages/prisma/prisma/schema.prisma [hunk 2]
```diff
@@ -22,7 +22,7 @@ model Account {
 
 model Session {
   id           Int      @id @default(autoincrement())
```
Comment (kripod) [inline packages/prisma/prisma/schema.prisma:24]: I think all the IDs should be strings for the sake of uniformness.

**Review by kripod — APPROVED** [review]
Looks good to me 😊

I would like to suggest using string IDs for `Account` and `Session` inside `schema.prisma`.

File: packages/prisma/prisma/schema.prisma (hunk 1 above)
Comment (balazsorban44) [reply packages/prisma/prisma/schema.prisma]: Done

File: packages/prisma/prisma/schema.prisma (hunk 2 above)
Comment (balazsorban44) [reply packages/prisma/prisma/schema.prisma]: Done

**Review by kripod — APPROVED** [review]

**Review by williamluke4 — APPROVED** [review]

**Pull request merged**